infra/terraform/terraform.go

```diff
@@ -64,45 +71,73 @@ func NewFromState(config Config, stateConfig infra.ProvisionerState) (*terraform
 }
 
 func (r *terraform) Create(withInstaller bool) (installer infra.Node, err error) {
-	err = system.CopyFile(filepath.Join(r.stateDir, "terraform.tf"), r.ScriptPath)
-	if err != nil {
-		return nil, trace.Wrap(err)
-	}
+	ctx, cancel := context.WithTimeout(context.Background(), terraformTimeout)
```

Comment:
that would be a chain reaction; it'll have to go top-down between most functions in infra, framework and lib package; which is fine but it should be subject to separate PR, likely coupled with changes to more fine-grained SSH command execution. 

please approve as-is.